components/image-builder/pkg/builder/docker.go

```diff
@@ -224,6 +224,10 @@ func (b *DockerBuilder) getBaseImageRef(ctx context.Context, bs *api.BuildSource
 		return b.getAbsoluteImageRef(ctx, src.Ref.Ref, allowedAuth)
 
 	case *api.BuildSource_File:
+		if src.File.DockerfileContentSha != "" {
+			return fmt.Sprintf("%s:%x", b.Config.BaseImageRepository, src.File.DockerfileContentSha), nil
```

Comment:
The manifest approach below takes more into account than just the Dockerfile version. Using just the content hash ignores the context entirely - the same Dockerfile from two separate repositories with different context would get the same "build".

This could even leak sensitive details because one user might get the build of another (same Dockerfile content, but different context including repo).

Reply:
Note: because GitHub doesn't let me put a comment in line 238 ... you'd have to expand the `if` to include the `CompositeInitializer` etc.